File: tests/test10/equi_filter.py
#!/usr/bin/python3


import sys


lines = []
for line in sys.stdin:
	lines.append(line)


for line in lines:
	words = line.split()

	
	for word in words:
		chars = {}
		
		# index character count 
		for char in tuple(word):
			if char.lower() in chars:
				chars[char.lower()] += 1
			else:
				chars[char.lower()] = 1
	
		# delete spaces
		if '' in chars:
			del chars['']
		# set standard
		val = list(chars.values())[0]
		
		print_word = True

		for char in chars.keys():
			if chars[char] != val:
				print_word = False


		if print_word == True:
			print(word, end=" ")




	print()

File: tests/test10/equi_filter.pl
#!/usr/bin/perl
my @lines;

while (<>) {
	chomp;
	push @lines, $_;
}






foreach my $line (@lines) {
	@words = split ' ', $line;
	
	# loop through each word	
	foreach my $word (@words) {
		my %chars;

		@letters = split //, $word;
		# index characters
		foreach my $letter (@letters) { $chars{lc($letter)}++ }
		# delete spaces
		delete $chars{''};
		# set character count standard
		my $val = $chars{(keys %chars)[0]};
		
		# loop through each char and check if it 
		# complies to char count standard
		my $print_word = 1;
		foreach my $char (keys %chars) {
			if ($chars{$char} != $val) {
				$print_word = 0;
			}
		}
	
		if ($print_word == 1) {
			print "$word ";
		}

	}

	
	print "\n";

}
